src/theory/relevance_manager.cpp

```diff
@@ -156,7 +183,29 @@ void RelevanceManager::computeRelevance()
                            << std::endl;
     }
   }
-  d_success = true;
+  d_success = !d_fullEffortCheckFail;
+}
+
+bool RelevanceManager::computeRelevanceFor(TNode input)
+{
+  int32_t val = justify(input);
+  if (val != 1)
+  {
+    // if we are in full effort check and fail to justify, then we should
+    // give a failure and set success to false, or otherwise calls to
+    // isRelevant cannot be trusted.
+    if (d_inFullEffortCheck)
+    {
+      std::stringstream serr;
+      serr << "RelevanceManager::computeRelevance: WARNING: failed to justify "
+           << input;
+      Trace("rel-manager") << serr.str() << std::endl;
+      Assert(false) << serr.str();
```

Comment:
```suggestion
      Assert(false) << "RelevanceManager::computeRelevance: WARNING: failed to justify " << input;
```

Reply:
This changes the intended behavior, I want a debug assertion failure or a Trace warning in production.

Reply:
My concern is that this induces an overhead in a (non-tracing) production build.
How about this?
```
      Trace("rel-manager") << "RelevanceManager::computeRelevance: failed to justify " << input << std::endl
      Assert(false) << "RelevanceManager::computeRelevance: failed to justify " << input;
```

Reply:
This code really should never be run, so performance is not an issue here.